supervision/detection/utils.py

```diff
@@ -48,6 +48,36 @@ def polygon_to_mask(polygon: np.ndarray, resolution_wh: Tuple[int, int]) -> np.n
     return mask
 
 
+def box_iou(box1: np.ndarray, box2: np.ndarray) -> float:
+    """
+    Compute the Intersection over Union (IoU) between two bounding boxes.
```

Comment:
Please make the documentation of `box_iou` a little more detailed; ensure that a developer referring to the docs of `box_iou` or `box_iou_batch` knows when is it recommended to use either functions.